### PR TITLE
[SYCL] Fix and reenable unit test for xpti_trace

### PR DESCRIPTION
**Pull request by KseniyaTikhomirova**

### CODE REVIEW

File: sycl/unittests/xpti_trace/QueueApiFailures.cpp [hunk 1]
```diff
@@ -502,7 +507,9 @@ TEST_F(QueueApiFailures, DISABLED_QueueKernelAsync) {
 
   // Need to wait till host task enqueue kernek to check code location report.
   using namespace std::chrono_literals;
-  std::this_thread::sleep_for(10ms);
+  while (EnqueueKernelLaunchCalled != true)
+    std::this_thread::sleep_for(1ms);
```
Comment (sergey-semenov) [inline sycl/unittests/xpti_trace/QueueApiFailures.cpp:511]: Would it be better to use `std::condition_variable` here instead?

Comment (KseniyaTikhomirova) [reply sycl/unittests/xpti_trace/QueueApiFailures.cpp]: done

**Review by sergey-semenov — APPROVED** [review]

**Comment by KseniyaTikhomirova** [comment]
@intel/llvm-gatekeepers hello, this PR is ready for merge

**Comment by aelovikov-intel** [comment]
This PR has caused failures in post-commit: https://github.com/intel/llvm/actions/runs/5145726160/jobs/9263753413.

**Comment by KseniyaTikhomirova** [comment]
> This PR has caused failures in post-commit: https://github.com/intel/llvm/actions/runs/5145726160/jobs/9263753413.

uploaded PR to fix https://github.com/intel/llvm/pull/9698

**Pull request merged**